cpp/src_prims/distance/pairwise_distance_base.cuh

```diff
@@ -150,16 +151,33 @@ struct PairwiseDistances : public BaseClass {
   }
 
   DI void epilog() {
-    epilog_op(acc, sxNorm, syNorm);
+    if (distanceType <= raft::distance::DistanceType::CosineExpanded) {
```

Comment:
Instead of relying on the `distanceType`, how about relying on whether the `xNorm` and `yNorm` pointers are `nullptr` or not? So, if in future a particular distance metric needs the norm values, it just has to pass these pointers explicitly, while all other metrics can just pass a `nullptr`. What say?

Reply:
my concern with checking `xNorm/yNorm` is that every distance metric would need to do this check in spite of it not needing to enter this `if`, currently with `distanceType` been template argument these check happens at compile time hence eliminating that runtime check along with it eliminating register usage.
Also in future if some other distance metric is using this class then it can add its enum value to this `if` condition.

Reply:
my only concern is that people could easily miss such a thing while adding a new metric. However, I agree with your point, it's better not to use `xNorm`/`yNorm` pointer checks. What about having a boolean template parameter like `UseNorms` to explicitly specify if norms are required during the epilog phase?

Reply:
`UseNorms` sounds like a good plan, I will remove `distanceType` as template argument as it will now become redundant once we add `UseNorms` boolean parameter.

Reply:
`UseNorms` is now added instead of `distanceType`

Reply:
Thanks Mahesh